index.bs

```diff
@@ -8041,63 +8043,367 @@ ECMAScript Object values.
+            1.  Perform |steps| with |r|.
+</div>
+
+<div algorithm>
+    To <dfn export for=Promise lt="wait for all|waiting for all">wait for all</dfn> with a [=list=]
+    of <code>{{Promise}}&lt;|T|&gt;</code> values |promises|, with success steps |successSteps|
+    that take a [=list=] of |T| values and failure steps |failureSteps| that take a rejection
+    reason {{any}} value, perform the following steps:
+
+    1.  Let |fullfilledCount| be 0.
+    1.  Let |rejected| be false.
+    1.  Let |rejectionHandlerSteps| be the following steps given |arg|:
+        1.  If |rejected| is true, abort these steps.
+        1.  Set |rejected| to true.
+        1.  Perform |failureSteps| given |arg|.
+    1.  Let |rejectionHandler| be [=!=] [$CreateBuiltinFunction$](|rejectionHandlerSteps|, « »):
+    1.  Let |total| be |promises|'s [=list/size=].
+    1.  If |total| is 0, then:
+        1.  [=Queue a microtask=] to perform |successSteps| given « ».
+        1.  Return.
+    1.  Let |index| be 0.
+    1.  Let |result| be a [=list=] containing |total| null values.
+    1.  [=list/For each=] |promise| of |promises|:
+        1.  Let |promiseIndex| be |index|.
+        1.  Let |fulfillmentHandler| be the following steps given |arg|:
+            1.  Set |result|[|promiseIndex|] to |arg|.
+            1.  Set |fullfilledCount| to |fullfilledCount| + 1.
+            1.  If |fullfilledCount| equals |total|, then perform |successSteps| given |result|.
+        1.  Let |fulfillmentHandler| be [=!=] [$CreateBuiltinFunction$](|fulfillmentHandler|, « »):
+        1.  Perform [$PerformPromiseThen$](|promise|, |fulfillmentHandler|, |rejectionHandler|).
+        1.  Set |index| to |index| + 1.
+</div>
+
+This phrase is useful when you wish to aggregate the result of multiple promises, and react to them
+all together, in the same way that {{Promise/all()|Promise.all()}} functions for JavaScript code.
+
+<div algorithm>
+    To
+    <dfn id="waiting-for-all-promise" export lt="get a promise to wait for all|getting a promise to wait for all">get a promise for waiting for all</dfn>
+    with a [=list=] of <code>{{Promise}}&lt;|T|&gt;</code> values |promises| and a [=Realm=]
+    |realm|, perform the following steps:
+
+    1.  Let |promise| be [=Promise/a new promise=] of type
+        <code>{{Promise}}&lt;<a lt="sequence type">sequence</a>&lt;|T|>></code> in |realm|.
+    1.  Let |successSteps| be the following steps, given |results|:
+        1.  [=Promise/Resolve=] |promise| with |results|.
+    1.  Let |failureSteps| be the following steps, given |reason|:
+        1.  [=Promise/Reject=] |promise| with |reason|.
+    1.  [=Promise/Wait for all=] with |promises|, given |successSteps| and |failureSteps|.
+    1.  Return |promise|.
+</div>
+
+This phrase is useful when you wish to aggregate the results of multiple promises, and then produce
+another promise from them.
+
+
+<h6 id="es-promise-examples">Examples</h6>
```

Comment:
My intention with the "add a heading" comment was for this to stay a h5, i.e. have two siblings under the promise section. Arguably this section has examples of the general promise type definitions too.